src/pymor/discretizations/iosys.py

```diff
@@ -0,0 +1,1702 @@
+        E = load_matrix(files_basename + '.E') if os.path.isfile(files_basename + '.E') else None
+
+        return cls.from_matrices(A, B, C, D, E, cont_time=cont_time,
+                                 solver_options=solver_options, estimator=estimator, visualizer=visualizer,
+                                 cache_region=cache_region, name=name)
+
+    def __mul__(self, other):
+        """Multiply (cascade) two |LTISystems|."""
+        assert self.B.source == other.C.range
+
+        A = BlockOperator([[self.A, Concatenation(self.B, other.C)],
+                           [None, other.A]])
+        B = BlockOperator.vstack((Concatenation(self.B, other.D),
+                                  other.B))
+        C = BlockOperator.hstack((self.C,
+                                  Concatenation(self.D, other.C)))
+        D = Concatenation(self.D, other.D)
+        E = BlockDiagonalOperator((self.E, other.E))
+
+        return self.__class__(A, B, C, D, E, cont_time=self.cont_time)
+
+    @cached
+    def poles(self, force_dense=False):
+        """Compute system poles.
+
+        Parameters
+        ----------
+        force_dense
+            Should dense matrix computations be forced.
+        """
+        if not force_dense:
+            if not (isinstance(self.A, NumpyMatrixOperator) and not self.A.sparse):
+                raise TypeError('Expected A to be NumpyMatrixOperator with dense matrix (force_dense == False).')
+            if not ((isinstance(self.E, NumpyMatrixOperator) and not self.E.sparse) or
+                    _is_like_identity_operator(self.E)):
+                raise TypeError('Expected E to be NumpyMatrixOperator with dense matrix or IdentityOperator'
+                                '(force_dense == False).')
+
+        A = to_matrix(self.A, format='dense')
+        E = None if _is_like_identity_operator(self.E) else to_matrix(self.E, format='dense')
+        return spla.eigvals(A, E)
+
+    def eval_tf(self, s):
+        """Evaluate the transfer function.
+
+        The transfer function at :math:`s` is
+
+        .. math::
+            C (s E - A)^{-1} B + D.
+
+        .. note::
+            We assume that either the number of inputs or the number of
+            outputs is much smaller than the order of the system.
+
+        Parameters
+        ----------
+        s
+            Complex number.
+
+        Returns
+        -------
+        tfs
+            Transfer function evaluated at the complex number `s`,
+            |NumPy array| of shape `(self.p, self.m)`.
+        """
+        A = self.A
+        B = self.B
+        C = self.C
+        D = self.D
+        E = self.E
+
+        sEmA = LincombOperator((E, A), (s, -1))
+        if self.m <= self.p:
+            tfs = C.apply(sEmA.apply_inverse(B.as_range_array())).to_numpy().T
+        else:
+            tfs = B.apply_adjoint(sEmA.apply_inverse_adjoint(C.as_source_array())).to_numpy().conj()
+        if not _is_like_zero_operator(D):
```

Comment:
Maybe it would be better to check this once and for all in `__init__`, convert `D` to a ZeroOperator if needed and only check `isinstance(D, ZeroOperator)` in the following?

Reply:
The problem appears when adding or subtracting two systems, or when converting from second-order to first-order. The resulting `D` operator is then a 1x1 `BlockOperator` of a `ZeroOperator`. So, as far as I see, the alternative is modifying `InputOutputSystem.__add__` and `SecondOrderSystem.to_lti`. Or somehow making `BlockOperator([[ZeroOperator]])` return a `ZeroOperator`.

Reply:
Yes, they do. Do you mean that `LTISystem.__init__` could convert `BlockOperator([[ZeroOperator]])` to a `ZeroOperator`?

Reply:
Yes, that was the idea. However, as said [above](https://github.com/pymor/pymor/pull/464#discussion_r235022078), if these cases can be easily avoided by a simple modification of `__add__` and `to_lti`, then this might be the cleanest option.